daemon/main.c

```diff
@@ -1065,37 +1082,37 @@ static void get_netdata_configured_variables() {
     // ------------------------------------------------------------------------
     // get default database update frequency
 
-    default_rrd_update_every = (int) config_get_number(CONFIG_SECTION_DB, "update every", UPDATE_EVERY);
-    if(default_rrd_update_every < 1 || default_rrd_update_every > 600) {
-        netdata_log_error("Invalid data collection frequency (update every) %d given. Defaulting to %d.", default_rrd_update_every, UPDATE_EVERY);
-        default_rrd_update_every = UPDATE_EVERY;
-        config_set_number(CONFIG_SECTION_DB, "update every", default_rrd_update_every);
+    rrdb.default_update_every = (int) config_get_number(CONFIG_SECTION_DB, "update every", rrdb.default_update_every);
+    if(rrdb.default_update_every < UPDATE_EVERY_MIN || rrdb.default_update_every > UPDATE_EVERY_MAX) {
+        netdata_log_error("Invalid data collection frequency (update every) %d given. Defaulting to %d.", rrdb.default_update_every, UPDATE_EVERY_MIN);
+        rrdb.default_update_every = UPDATE_EVERY_MIN;
+        config_set_number(CONFIG_SECTION_DB, "update every", rrdb.default_update_every);
     }
 
     // ------------------------------------------------------------------------
-    // get default memory mode for the database
+    // get default storage engine for the database
 
     {
-        const char *mode = config_get(CONFIG_SECTION_DB, "mode", rrd_memory_mode_name(default_rrd_memory_mode));
-        default_rrd_memory_mode = rrd_memory_mode_id(mode);
-        if(strcmp(mode, rrd_memory_mode_name(default_rrd_memory_mode)) != 0) {
-            netdata_log_error("Invalid memory mode '%s' given. Using '%s'", mode, rrd_memory_mode_name(default_rrd_memory_mode));
-            config_set(CONFIG_SECTION_DB, "mode", rrd_memory_mode_name(default_rrd_memory_mode));
+        const char *se_name = config_get(CONFIG_SECTION_DB, "mode", storage_engine_name(default_storage_engine_id));
+        if (!storage_engine_id(se_name, &default_storage_engine_id)) {
+            netdata_log_error("Invalid memory mode '%s' given. Using '%s'",
+                              se_name, storage_engine_name(default_storage_engine_id));
+            config_set(CONFIG_SECTION_DB, "mode", storage_engine_name(default_storage_engine_id));
```

Comment:
You renamed our variable from `mode` to `id` in our source code,

``` sh
for(size_t tier = 0; tier < storage_tiers ;tier++) {
                    if(host->db[tier].id != STORAGE_ENGINE_DBENGINE) continue;
                    if(!host->db[tier].instance) continue;
````

but we are still reading `mode` from config file. Is really necessary to rename `mode` -> `id`? Do you also plan to modify the user option?

Reply:
> but we are still reading mode from config file. Is really necessary to rename mode -> id? 

Yes, `id` is used to identify the storage engine we want to use and there's no code anymore that uses the `_MODE` prefix.

> Do you also plan to modify the user option?

No, but I'd be happy to do so if we want. My main goal is to have a single consistent way of naming stuff internally in our code. Users and the documentation can still refer to "memory modes".

Reply:
All right, let us discuss this in our meeting that @ktsaou suggested.